test/modules/tiledimage.js

```diff
@@ -205,10 +205,13 @@
 
             viewer.addHandler('tiled-image-drawn', function updateHandler2() {
                 viewer.removeHandler('tiled-image-drawn', updateHandler2);
-                assert.ok(viewer.tileCache.numTilesLoaded() > 0, 'more tiles load');
-                viewer.world.getItemAt(0).destroy();
-                assert.equal(viewer.tileCache.numTilesLoaded(), 0, 'no tiles after destroy');
-                done();
+
+                setTimeout(() => {
```

Comment:
One of examples how async changes the behavior. Sometimes, a frame was fired sooner than any tile got loaded (since now there is some job to do -> convert blob to image or url to image, and this takes more time and does not block the rendering - there was not immediatelly something loaded). Question is whether we want to tackle this somehow, but I would also say good that we make stuff more flexible and the frames are not blocked that much.

Reply:
Agreed.

Do you find it has any effect on the perceived loading time?

Reply:
I tested this on the drawer performance with 20 overlays. I would say it behaves as expected. Although the main 'culprit' here is the extra `setTimeout`, not changes of where transfer to target data type happens. 

- Commented out  `setTimeout` / master branch: layers load individually, but a bit faster than in the second case. At the cost of stuttering animation, the viewer animation lags while it's loading 20 overlays.
- Active  `setTimeout`: layers load individually, they appear in bulks after some time. Animations are bit smoother.

But sometimes I could not really tell the difference. Looking at profiler, `setTimeout` generally makes frame handler bit faster...

Reply:
Cool. Well, getting rid of stutter is definitely worth while!